### PR TITLE
Implement asynchronous find function.

### LINKED ISSUE #3: Implement asynchronous find
Function `metasync.find(items, iterator, done);` is an asynchronous analogue for synchronous `Array.find`

### PR DESCRIPTION
**Pull request by anxolerd**
Implement asynchronous find function. Implemented function accepts 3 arguments:

 * predicate
 * array of data
 * callback

It asynchronously applies predicate to each element in array and if predicate returns `true`, it calls the callback with the found element as its only argument. If the function reaches the end of the array and the element has not been found, the callback is called without any arguments (which means all arguments are `undefined`)

The example of using `metasync.find` was added. The example uses asynchronous `find` function to find a value which is divisible by 3 and 5 simultaneously.

Resolves #3

### CODE REVIEW

File: metasync.js [hunk 1]
```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
   if (len > 0) next();
   else done();
 };
+
+
+// Asynchronous find function
+// find :: (a -> Boolean) -> [a] -> (a -> Void)
+metasync.find = function(predicate, array, done) {
```
Comment (tshemsedinov) [inline metasync.js]: better metasync.find(array, predicate, done)

File: metasync.js [hunk 2]
```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
...
+        index++;
+
+        // Use setTimeout to yield
+        setTimeout(doFind, 0);
+      }
+    }
+  }
+
+  // Use setTimeout to yield
+  setTimeout(doFind, 0);
```
Comment (tshemsedinov) [inline metasync.js]: remove setTimeout

File: metasync.js [hunk 3]
```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
...
+    if (index === array.length) {
+      done()
+    } else {
+      if (predicate(array[index])) {
+        done(array[index]);
+      } else {
+        index++;
+
+        // Use setTimeout to yield
+        setTimeout(doFind, 0);
```
Comment (tshemsedinov) [inline metasync.js]: remove setTimeout

File: metasync.js [hunk 4]
```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
...
+// Asynchronous find function
+// find :: (a -> Boolean) -> [a] -> (a -> Void)
+metasync.find = function(predicate, array, done) {
+  var index = 0;
+
+  function doFind() {
+    if (index === array.length) {
+      done()
+    } else {
+      if (predicate(array[index])) {
```
Comment (tshemsedinov) [inline metasync.js]: predicate should return using callback

**Comment by anxolerd** [comment]
I'll fix this right after #4, as the problems are common.

**Pull request merged**